quickjs-libc.c

```diff
@@ -3910,54 +3910,52 @@ JSModuleDef *js_init_module_os(JSContext *ctx, const char *module_name)
 
 /**********************************************************/
 
-#ifdef _WIN32
-static JSValue js_print(JSContext *ctx, JSValue this_val,
-                        int argc, JSValue *argv)
-{
-    int i;
-    const char *str;
-    size_t len;
-    DWORD written;
-    HANDLE hConsole = GetStdHandle(STD_OUTPUT_HANDLE);
-    if (hConsole == INVALID_HANDLE_VALUE)
-        return JS_EXCEPTION;
-    for(i = 0; i < argc; i++) {
-        if (i != 0)
-            WriteConsoleW(hConsole, L" ", 1, &written, NULL);
-        str = JS_ToCStringLen(ctx, &len, argv[i]);
-        if (!str)
-            return JS_EXCEPTION;
-        DWORD prev = GetConsoleOutputCP();
-        SetConsoleOutputCP(CP_UTF8);
-        WriteConsoleA(hConsole, str, len, &written, NULL);
-        SetConsoleOutputCP(prev);
-        JS_FreeCString(ctx, str);
-    }
-    WriteConsoleW(hConsole, L"\n", 1, &written, NULL);
-    FlushFileBuffers(hConsole);
-    return JS_UNDEFINED;
-}
-#else
 static JSValue js_print(JSContext *ctx, JSValue this_val,
                         int argc, JSValue *argv)
 {
+#ifdef _WIN32
+    HANDLE handle;
+    DWORD mode;
+#endif
+    const char *s;
+    DynBuf b;
     int i;
-    const char *str;
-    size_t len;
+
+    dbuf_init(&b);
     for(i = 0; i < argc; i++) {
-        if (i != 0) 
-            putchar(' ');
-        str = JS_ToCStringLen(ctx, &len, argv[i]);
-        if (!str)
-            return JS_EXCEPTION;
```

Comment:
Functional change which IMO is an improvement: exceptions no longer cause truncated output